Adding additinal tv channel card to the media player modelUpdate

miloit opened the pull request:
Adding additinal tv channel card to the media player entity
Adding EPG View Card and EPGItem Model to media player entity
Reorganisation of Bottom Menu in Media Player Entity

miloit committed:
Update

miloit committed:
New EPG Model implementation

miloit committed:
Code cleaning

martonborzak commented on components/media_player/ui/CardMediaPlayerEPGView.qml:
```diff
@@ -0,0 +1,253 @@
+                        Text {
+                            id: title
+                            color: Style.color.text
+                            text: "TV Kanal"//qsTr("My playlists") + translateHandler.emptyString
```
Please use `qsTr` to mark this for translation. Default language is English.

martonborzak commented on components/media_player/ui/CardTVChannelList:
```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+ *
+ * Copyright (C) 2018-2019 Marton Borzak <hello@martonborzak.com>
```
Not me :)

miloit replied:
Tha's strange I see there my name......nevertheless I pushed it again

martonborzak commented on components/media_player/ui/CardTVChannelList.qml:
```diff
@@ -0,0 +1,253 @@
+                        Text {
+                            id: title
+                            color: Style.color.text
+                            text: "TV Kanal"//qsTr("My playlists") + translateHandler.emptyString
```
Please use `qsTr` to mark this for translation. Default language is English.

miloit replied:
solved

martonborzak commented on components/remote/ui/ir/CardButtons.qml:
```diff
@@ -23,7 +23,8 @@ import QtQuick 2.11
 import Style 1.0
 
 import Haptic 1.0
-import Entity.Remote 1.0
```
I don't think this should be removed.

miloit replied:
of course not....

martonborzak commented on remote.pro:
```diff
@@ -506,6 +506,11 @@ win32 {
 
 DISTFILES += \
     components/FpsItem.qml \
+    components/media_player/ui/CardMediaPlayerEPGView.qml \
+    components/media_player/ui/CardMediaPlayerEPGView.qml \
+    components/media_player/ui/CardMediaPlayerRemote.qml \
+    components/media_player/ui/EPGItemView.qml \
+    components/media_player/ui/MediaPlayerRemoteView.qml \
```
These should not be added here. These are in a resource file.

miloit replied:
solved

miloit committed:
changes according review

miloit committed:
changes accoring review

miloit committed:
Code Style

miloit committed:
Update dependencies.cfg

Update dependencies

miloit committed:
Delete CardTVChannelList

Deleted wrong file

zehnm commented on hardware-schema.json:
```diff
@@ -141,7 +143,10 @@
         "enabled": { "$ref": "#/definitions/enabled" },
         "i2c":  { "$ref": "#/definitions/i2c" },
         "interrupt": {
-          "gpio":  { "$ref": "#/definitions/gpio" }
```
good catch!

miloit replied:
Sorry, didn't unerstand it.....Can you explain me?

zehnm replied:
In JSON the gpio node is a child node of interrupt: `interrupt/gpio`
The corresponding schema definition was wrong, the object definition was missing which defines that that gpio object has property objects.

zehnm commented on config-schema.json:
```diff
@@ -921,7 +932,6 @@
             },
             "updateUrl": {
               "type": "string",
-              "format": "url",
```
The format specifier may not be removed. It makes sure, that the string conforms to an URL and that the user can't enter an invalid format.

An issue might be that `url` is not supported by all json tools, since that's an area of proposals and recommendations instead of well defined specifications...
The ajv validator supports it, others may not. As I just found out, that the cpp validator we are using just ignores it! Therefore I'll change the format to the following alternative:

```json
{
    "type": "string",
    "format": "uri",
    "pattern": "^(https?)://"
}
```

zehnm replied:
Ok doesn't work anymore :-( ajv was happy last year with both uri and url formats!
This is exactly why I don't like JSON and JavaScript. No standard, suddenly after a tool or dependency update it's no longer working...

I'll replace it with a pure regex then

zehnm replied:
Reason: ajv put the formats in an external library
https://ajv.js.org/docs/validation.html#formats
Now: how to use that in the command line version?

zehnm replied:
Fixed in master branch: ajv-cli version downgraded to 3.3.0

zehnm reviewed (approved):
Since I'm not a QML guy, I'd say: looks good to me :-)

I will change the json url format specifier directly into this PR to make sure it's not removed.

zehnm committed:
fix: URL pattern definition in config JSON schema

The `url` format specifier isn't supported by all tools.
Use the simpler `uri` format with a pattern restriction instead.

zehnm committed:
Why can't there be one json schema standard?

zehnm committed:
Back to uri format

Schema validation has been fixed in master branch

zehnm commented:
@martonborzak can I squash and merge this PR, or is there still something missing?

martonborzak commented on components/media_player/ui/Card.qml:
```diff
@@ -205,119 +285,221 @@ Rectangle {
-        }
-    }
+
+        }}
```
I know it's nitpicking, but can we break these into two lines and indent them correctly? :) Thanks

martonborzak reviewed (approved):
I think it's good! Thanks @miloit. There's only a very small thing and it's not even important for this PR. Just cosmetics :)

zehnm committed:
format fix
